backend/src/neuroagent/tools/circuit_population_analysis_tool.py

```diff
@@ -0,0 +1,335 @@
+
+It converts natural language questions about neural circuit populations into SQL queries and executes them
+against the population DataFrame. The tool supports comprehensive analysis of neuron populations following
+the SONATA data format specification, including:
+
+- Filtering neurons by spatial properties (3D coordinates, layer, region, subregion)
+- Analyzing cell types (biophysical, point_neuron, single_compartment, virtual)
+- Examining morphological properties (mtype, morphology files, model templates)
+- Investigating electrical properties (etype, excitatory/inhibitory classification)
+- Statistical analysis of population distributions and characteristics
+- Spatial queries for neuron positioning and circuit topology
+
+The tool understands SONATA-specific terminology and data structures, including node types, model types,
+morphological classifications, and circuit organization principles.
+
+Input:
+- circuit_id: UUID of the circuit
+- sonata_asset_id: UUID of the compressed sonata circuit data
+- population_name: Name of the neural population to analyze, it will only keep this particular population.
+- question: A natural language question about the neurons in the population, DO NOT MENTION the population name in the question.
+
+Output: Analysis results showing neuron data based on the query, formatted according to SONATA standards
+"""
+
+    description_frontend: ClassVar[
+        str
+    ] = """Analyze SONATA neural circuit populations using natural language.
+Ask questions like "What is the most common morphological type?", "How many excitatory neurons are in layer 5?", "Show me all biophysical neurons in visual cortex", or "What morphologies are used by inhibitory cells?" and get detailed circuit analysis."""
+    metadata: CircuitPopulationAnalysisMetadata
+    input_schema: CircuitPopulationAnalysisInput
+
+    async def _download_and_extract_circuit(self, temp_dir: str) -> str:
+        """Download and extract circuit data, return path to config file."""
+        # Get pre-signed url
+        headers: dict[str, str] = {}
+        if self.metadata.vlab_id is not None:
+            headers["virtual-lab-id"] = str(self.metadata.vlab_id)
+        if self.metadata.project_id is not None:
+            headers["project-id"] = str(self.metadata.project_id)
+
+        response = await self.metadata.httpx_client.get(
+            url=f"{self.metadata.entitycore_url.rstrip('/')}/circuit/{self.input_schema.circuit_id}/assets/{self.input_schema.sonata_asset_id}/download",
+            headers=headers,
+            follow_redirects=False,
+        )
+        if response.status_code != 307:
+            raise ValueError(
+                f"The asset download endpoint returned a non 307 response code. Error: {response.text}"
+            )
+        presigned_url = response.headers["location"]
+
+        # Download the .gz file
+        print("DOWNLOADING FILE")
+        os.makedirs(temp_dir, exist_ok=True)
```

Comment:
A detail, throughout the codebase we use `pathlib` to do anything path related. I would stick to that convention.